integration-tests/google-pubsub/src/main/java/org/apache/camel/quarkus/component/google/pubsub/it/GooglePubsubResource.java

```diff
@@ -64,7 +82,95 @@ public Response sendPojoToTopic(String fruitName) {
     @Produces(MediaType.APPLICATION_JSON)
     public Response consumePojoFromTopic() {
         Object response = consumerTemplate
-                .receiveBody("google-pubsub:{{project.id}}:{{subscription.name}}?synchronousPull=true", 5000L);
+                .receiveBody("google-pubsub:{{project.id}}:{{google-pubsub.subscription-name}}?synchronousPull=false", 5000L);
         return Response.ok(response).build();
     }
+
+    @Path("/sendToEndpoint")
+    @POST
+    public Response sentToEndpoint(String message,
+            @QueryParam("toEndpoint") String toEndpoint)
+            throws Exception {
+        producerTemplate.sendBody(toEndpoint, message);
+        return Response.created(URI.create("https://camel.apache.org")).build();
+    }
+
+    @Path("/getFromEndpoint")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getSchedulerResult(@QueryParam("fromEndpoint") String fromEndpoint) throws Exception {
+        return consumerTemplate.receiveBody(fromEndpoint, 5000, String.class);
+    }
+
+    @Path("receive/subscription/{subscriptionName}")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String receiveFromSubscription(@PathParam("subscriptionName") String subscriptionName) throws Exception {
+        return consumeEndpoint(subscriptionName, null);
+    }
+
+    @Path("receive/subscriptionOrdering/{subscriptionName}")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String receiveFromSubscriptionOrdered(@PathParam("subscriptionName") String subscriptionName) throws Exception {
+
+        return consumeEndpoint(subscriptionName,
+                "?messageOrderingEnabled=true&pubsubEndpoint=us-east1-pubsub.googleapis.com:443");
```

Comment:
So, this url will never change ?

Reply:
Yes. See https://cloud.google.com/pubsub/docs/reference/service_apis_overview#service_endpoints

Reply:
explained

Reply:
Thanks for the link @JiriOndrusek . Any reason why we specifically target us-east-1 ? Why not europe of global for instance ?

Reply:
I chose `us-east1-pubsub` to stay consistent with camel integration test - see https://github.com/apache/camel/blob/main/components/camel-google/camel-google-pubsub/src/test/java/org/apache/camel/component/google/pubsub/integration/MessageOrderingIT.java#L46

The condition for pubsub ordering feature is, that all messages have to be published in the same region to ensure ordering. I don't see a problem using global or different endpoint in the test.  @aldettinger  Which one do you suggest?

Reply:
The documentation mention 3 reasons for using regional endpoints. In this PR, my understanding is that we deal with small volumes, producers/subscribers in same region, without constraints on data locality. If one agree with this, then it would be tempting to use global so that tests run on a nearby available region. Maybe tests would run quicker and fail less frequently.

That said, it may not be that simple. Maybe we could try global and integrate in the PR only of it works straight forward ? Maybe it's the subject of another ticket ? Or maybe requesting to the global endpoint is not a good idea for some reasons ?

What do you think ?

Reply:
@aldettinger  You are right about the global, that makes more sense. I tried it and the configuration works without an issue.So I changed the PR to ise global, if you agree please mark this conversation resolved.

Reply:
In the PR, we still use `us-east1-pubsub.googleapis.com` ? Isn't it regional ? Would public endpoint be `us-east1-pubsub.googleapis.com` ?

Reply:
There were 2 occurrences of the `us-east1`, I fixed also the second one.